code/modules/multiz/movement/turf/turf_zfall.dm

```diff
@@ -53,17 +53,25 @@
 	if(!can_start_zFall(A, target, force, from_zfall))
 		return FALSE
 	if(from_zfall) // if this is a >1 level fall
-		sleep(2) // This is the delay between falling zlevels. Otherwise zfalls would be instant to the client, which does not look great.
-		var/turf/new_turf = get_turf(A) // make sure we didn't move onto a solid turf, if we did this will perform a zimpact via the caller
-		target = get_step_multiz(new_turf, DOWN)
-		if(!new_turf.can_start_zFall(A, target, force, from_zfall))
-			new_turf.do_z_impact(A, levels - 1)
-			return TRUE // skip parent zimpact - do a zimpact on new turf, the turf below us is solid
-		else if(new_turf != src) // our fall continues... no need to check can_start_zFall again, because we just checked it
-			new_turf.zFall_Move(A, levels, old_loc, target)
-			return TRUE // don't do an impact from the parent caller. essentially terminating the old fall with no actions
+		addtimer(CALLBACK(src, PROC_REF(zFall_Finish), A, levels, force, old_loc, from_zfall), 0.2 SECONDS) // This is the delay between falling zlevels. Otherwise zfalls would be instant to the client, which does not look great.
+	else
```

Comment:
This else is a way of making it run through the timer first, like how the code was originally. Otherwise we will have the person fall x levels and 1 level at the same time.

![image](https://github.com/user-attachments/assets/e5f5e521-39eb-429d-8192-e209a94593e0)